torch/fx/passes/infra/partitioner.py

```diff
@@ -110,13 +104,17 @@ def dfs_find_cycle(node):
 
             return True
 
-        def merge_single_node(node: Node, id: int):
-            assert node not in assignment
+        def merge_single_node(node: Node, id: Optional[int]):
+            if node in assignment:
+                partitions_by_id[assignment[node]].remove_node(node)
 
-            assignment[node] = id
-            if id not in partitions_by_id:
+            if id is None:
```

Comment:
Do we really have case where id is None？ 
Looking at the call site for this function, it doesn't seem to have any None case... 

If this cannot be None, let assert id is not None.

Reply:
Yes, this happens naturally when `getitem` is marked as `supported` by backends.

Later during the special handling of `getitem`, where we merge each `getitem` calls to its producer, we could run into cases where the producer is not supported by backends, but we accidentally merged `getitem` into the fusion. We'll ended up pulling these nodes out and that's when `id` here would be `None`.

https://github.com/pytorch/pytorch/blob/7d3fbd7603adcb0421373f5b1924b5adfd8f574a/torch/fx/passes/infra/partitioner.py#L167

This is checked in test case `forward13`